contracts/mesh-provider/src/ibc.rs

```diff
@@ -139,6 +139,12 @@ pub fn receive_rewards(
             None => return Err(ContractError::ValidatorRewardsCalculationWrong {}),
         };
 
+        let total_staked = val.shares_to_tokens(val.stake);
+
+        if total_staked.is_zero() {
+            return Err(ContractError::ValidatorRewardsCalculationWrong {});
```

Comment:
Seems like the error message could be more specific here? Would expect a "no tokens staked" error.